pkg/serviceaccount/legacy.go

```diff
@@ -148,6 +150,13 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 		// Track secret-based long-lived service account tokens and add audit annotations and metrics.
 		autoGenerated := false
 
+		// Check if the secret has been marked as invalid
+		if invalidSince, _ := secret.Labels[InvalidSinceLabelKey]; invalidSince != "" {
+			klog.Errorf("The autogenerated token has been marked as invalid since the date: %s, please use tokens from the TokenRequest API or manually created secret-based tokens. If you still want to use this token, please update the secret by removing the 'kubernetes.io/legacy-token-invalid-since' label line in the secret and try it again.", invalidSince)
+			v.patchSecretWithLastUsedDate(ctx, secret)
```

Comment:
we might not want to patch with last used if we decide to return an error.

Reply:
Removed the patch.